eql/parser.py

```diff
@@ -595,7 +598,7 @@ def _field_path(self, node, allow_optional=False):
                 name = to_unicode(part["NAME"])
                 full_path.append(name)
 
-                if name in keywords:
+                if name in keywords and ".".join(map(str, full_path)) not in as_keyword_exceptions:
```

Comment:
personally, I would just remove `as` as a keyword altogether. seems a little easier to maintain, and more consistent with how Elasticsearch handles it

Reply:
hmm, I think you want to double check with Endpoint first because Lark doesn't have keywords builtin, so I had to catch collisions here manually. but Elasticsearch and other implementations use ANTLR and may have different behavior. ultimately, when python EQL is parsed in Endpoint mode, you want to make sure you're mirroring that behavior as close as possible.

also -- you can make some keywords conditional on the flags here as well. so you could do something like:
```python
                if name in keywords or (self._endpoint_syntax_enabled and name in endpoint_keywords):
```

Reply:
Removing since in endpoint `as` only happens in a sequence, outside the `[]` and isn't reserved as a keyword.